conduit/telemetry/telemetryCommon.go

```diff
@@ -0,0 +1,41 @@
+package telemetry
+
+import (
+	"time"
+
+	"github.com/opensearch-project/opensearch-go"
+)
+
+// Config represents the configuration of Telemetry logging
+type Config struct {
+	Enable    bool
+	SendToLog bool
+	URI       string
+	Name      string
+	GUID      string
+	Index     string
+	UserName  string
+	Password  string
+}
+
+// Event represents a single event to be emitted to OpenSearch
+type Event struct {
+	// Time at which the event was created
+	Time time.Time `json:"timestamp"`
+	// Event message
+	Message string `json:"message"`
+	// ID
+	GUID string `json:"guid"`
+}
+
+// Client represents the Telemetry client and config
+type Client interface {
+	MakeTelemetryStartupEvent() Event
+	SendEvent(event Event) error
+}
+
+// OpenSearchClient holds the OpenSearch client and TelemetryConfig
+type OpenSearchClient struct {
```

Comment:
Oh! It already is wrapped. Could you rename this to TelemetryClient?

Reply:
I could, but the current lint configurations output an error because of stuttering:

```
exported: type name will be used as telemetry.TelemetryClient by other packages, and that stutters; consider calling this Client (revive)
    type TelemetryClient struct {
```

Should I disable this check and rename to `TelemetryClient` anyway? I'm okay either way.

Reply:
> I always forget about this. telemetry.Client seems good too, how about just Client?

The interface is named `Client` so there would be a collision unfortunately :(

Reply:
In that case outside of this package you can reference the interface instead of `telemetry.OpenSearchClient`

Reply:
Good point! Will do this